gitsync/sync/fetch_pipeline.py

```diff
@@ -39,6 +39,37 @@ def on_update(self, data: enc.BinaryStr, respond_to: typ.Optional[bytes]):
         if not self.in_process:
             self.in_process = True
             aio.create_task(self.after_update(ref_updates, respond_to))
+    
+    async def after_bmeta_commit(self, ref_updates: typ.Dict[str, bytes], force: bool=False):
+        # Convert refs/bmeta/* to refs/head/*
+        def head_name_of_bmeta(ref_name):
+            ref_name = ref_name.split('/')
+            ref_name[1] = 'head'
+            return '/'.join(ref_name)
+            
+        for name, head in ref_updates.items():
+            # Extract HEAD from HeadRef
+            wire = self.repo.get_commit(head).tree['head.tlv'].data_stream.read()
+            headref, _ = proto.parse(wire)
+            if not isinstance(headref, proto.HeadRef):
+                logging.error(f'File head.tlv is not of type HeadRef')
+                continue
+            head = headref.head
+
+            # Fetch local or remote objects from HEAD
+            try:
+                await self.fetcher.fetch('commit', head)
+            except (ValueError, InterestCanceled, InterestTimeout, InterestNack) as e:
+                logging.warning(f'Fetching error - {type(e)} {e}')
+                continue
```

Comment:
`bmeta` and `head` should always be consistent. You need to roll back `bmeta` branch if you cannot move `head` forward.